test/StressTestMetrics/Program.cs

```diff
@@ -0,0 +1,99 @@
+        private static Random random = new Random();
+
+        public static void Main(string[] args)
+        {
+            long numberOfMetricWriters = Environment.ProcessorCount;
+            long maxWritesPerWriter = 10000000;
+            var writes = new long[numberOfMetricWriters];
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddSource("TestMeter")
+                .Build();
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
+
+            Parallel.Invoke(
+                () =>
+                {
+                    var watch = new Stopwatch();
+                    while (true)
+                    {
+                        long totalWrites = 0;
+                        for (int i = 0; i < numberOfMetricWriters; i++)
+                        {
+                            totalWrites += writes[i];
+                        }
+
+                        watch.Restart();
+                        Thread.Sleep(1000);
+                        watch.Stop();
+
+                        long newTotalWrites = 0;
+                        for (int i = 0; i < numberOfMetricWriters; i++)
+                        {
+                            newTotalWrites += writes[i];
+                        }
+
+                        var writesPerSec = (newTotalWrites - totalWrites) / (watch.ElapsedMilliseconds / 1000.0);
+                        Console.Title = $"Writes (Million/Sec): {writesPerSec / 1000000}";
+
+                        if (totalWrites > numberOfMetricWriters * 10000000)
+                        {
+                            break;
+                        }
+                    }
+                }, () =>
+                {
+                    Parallel.For(0, numberOfMetricWriters, i
+                    =>
+                    {
+                        Console.WriteLine($"Metric writer {i} started.");
+                        while (writes[i]++ < maxWritesPerWriter)
+                        {
+                            // 10 * 10 * 10 = 1000 unique combination is produced
+                            // which is well within the current hard-coded cap of
+                            // 2000.
+                            var tag1 = new KeyValuePair<string, object>("DimName1", dimensionValues[random.Next(0, 10)]);
+                            var tag2 = new KeyValuePair<string, object>("DimName2", dimensionValues[random.Next(0, 10)]);
+                            var tag3 = new KeyValuePair<string, object>("DimName3", dimensionValues[random.Next(0, 10)]);
+                            Counter.Add(100, tag1, tag2, tag3);
+                        }
+
+                        Console.WriteLine($"Metric writer {i} completed.");
+                    });
+                });
+
+            var rate = (double)(numberOfMetricWriters * 10000000) / (sw.ElapsedMilliseconds / 1000);
```

Comment:
I think this line has some minor issues:

* replace `10000000` with `maxWritesPerWriter`.
* `sw.ElapsedMilliseconds / 1000` will introduce rounding / precision issue.